shark_turbine/kernel/wave/constraints.py

```diff
@@ -79,6 +79,14 @@ def linearized_thread_id(self) -> IndexExpr:
         ]
         return sum([x * y for x, y in zip(thread_ids, threads_per_block)])
 
+    # Inline substitution for vector_size given index map. In the future we can add support for other members.
+    def _subs(self, index_map: dict[IndexSymbol, int]):
```

Comment:
Since this is a public facing member function of hardware constraint, we can drop the leading underscore? My understanding was that leading underscore is for functions we don't want to expose to other modules/files.

Reply:
I see I see, I was following the pytorch convention of doing `_` meaning it's doing an inline/on variable modification. i.e  `x._add(y) == x += y` while `c = x.add(y) == x+y`. What do you think? I can change it back to subs if you think that's better/more uniform with the overall styling. :)

Reply:
Cool, good to know. We haven't been following that convention so far, so I would probably stick with subs.

Reply:
Renamed to `subs_vector_shapes ` based on the other comment. :)

Reply:
thanks!